### ISSUE TITLE
Enchantments in customenchants.yml do not work

### ISSUE THREAD

**Issue by KNNI** [Auxilor/EcoEnchants#128]
**Describe the bug**
After much testing with customenchants.yml, all custom enchantments do not work. No matter which effect, no matter if with or without trigger or condition.
With EssentialsX or similar I can enchant the custom enchantments with /enchant id but with /enchantinfo id or name they are not to be found. They are also not listed in the Villager trades. Loot and Enchantmenttable I have not tested so extensively.

**To Reproduce**
The whole test I have tested before on an already set up server. After I felt myself stupid, I set up a new server and tested it with default configurations.
- installation of a paper server
- installation of basic plugins

_[About command /enchant]_
- enchant item with a custom enchantment
- notice no effect from the effect, ... 

_[Try to get information with /enchantinfo]_
- enter /enchantinfo id or name
- "Enchantment not found message"

_[Villager tests]_
- deactivate all enchants under the folder "enchants" (for easy testing as only vanilla & the self-made enchants should appear)
- notice that no custom enchantments show up in Villager Trades (Tested with librarians).

**Expected behavior**
That enchantments listed in customenchants.yml do something.

**Screenshots**
see Additional context

**Server Information (please complete the following information):**
_from a new server set up:_
server operating system:  Debian GNU/Linux 10 (buster)
```
java version "17.0.1" 2021-10-19 LTS
Java(TM) SE Runtime Environment (build 17.0.1+12-LTS-39)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.1+12-LTS-39, mixed mode, sharing)
```
Version Paper: git-Paper-176 (MC: 1.18.1) 
Server JVM Tags:
```
java -Xms2G -Xmx2G -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:-OmitStackTraceInFastThrow -XX:+AlwaysPreTouch -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=8 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=true -Daikars.new.flags=true -jar "/directory/file.jar"
```
Version eco: ~~6.20.4~~ 6.21.1
Version ecoenchants: ~~8.29.7~~ 8.29.9
other plugins: EcoEnchantsGUI, Essentials, PlaceholderAPI (with player & server expansion), ProtocolLib, StatTrackers, Vault, WorldEdit - All with default configs.

**Additional context**
i have set up a test server with these settings. if you want to see for yourself: knni.de:26000
in addition to this testserver i also created a customenchantment. otherwise everything is default.
If someone wants to have OP, just let me know. i try to watch the console to give the rights directly to those if necessary.
if anything needs to be configured, let me know as well.

the additional customenchantment:
```
  - id: test
    type: normal
    name: "test"
    description: test enchant desc
    obtaining:
      table: true
      villager: true
      loot: true
      rarity: common
    general-config:
      flags: [ ]
      targets:
        - all
      grindstoneable: true
      disabled-in-worlds: [ ]
      requirements:
        list: [ ]
        not-met-lore: [ ]
      conflicts: [ ]
    levels:
      - effects:
          - id: bonus_health
            args:
              health: 10
        conditions: [ ]
      - effects:
          - id: bonus_health
            args:
              health: 20
        conditions: [ ]
```

thanks for any help.

**Comment by KNNI** [comment]
hey! with ecoenchants version 8.29.10 and eco version 6.21.1 the effects, triggers etc. work for me!
but still no proper output is given at /enchantinfo id or name (description etc.) with EcoenchantsGUI the own enchantments are also not visible.
Likewise, I have also tested the Villager again. there I could also not find the self-created enchantments (30 minutes search with librarians). So I think that in the enchantmenttable and in the loot these are also not to find.

on the testserver (see additional context) i changed the customenchants.yml a little bit. also i deactivated all other enchantments except vanilla for the villager test.
complete customenchants.yml:

```
enchants:
  - id: escape
    type: normal
    name: "Escape"
    description: Gain a short burst of speed after taking damage

    obtaining:
      table: true
      villager: true
      loot: true
      rarity: epic

    general-config:
      flags: [ ]
      targets:
        - boots
      grindstoneable: true
      disabled-in-worlds: [ ]
      requirements:
        list: [ ]
        not-met-lore: [ ]
      conflicts: [ ]

    levels:
      - effects:
          - id: potion_effect
            args:
              effect: speed
              level: 1
              duration: 30
              apply_to_player: true
            triggers:
              - take_damage
        conditions: [ ]
      - effects:
          - id: potion_effect
            args:
              effect: speed
              level: 2
              duration: 30
              apply_to_player: true
            triggers:
              - take_damage
        conditions: [ ]
  - id: test
    type: normal
    name: "test"
    description: test enchant desc
    obtaining:
      table: true
      villager: true
      loot: true
      rarity: common
    general-config:
      flags: [ ]
      targets:
        - helmet
      grindstoneable: true
      disabled-in-worlds: [ ]
      requirements:
        list: [ ]
        not-met-lore: [ ]
      conflicts: [ ]
    levels:
      - effects:
          - id: bonus_health
            args:
              health: 10
        conditions: [ ]
  - id: enderbow
    type: normal
    name: "Enderbow"
    description: Teleport to the landing spot of your arrow
    obtaining:
      table: true
      villager: true
      loot: true
      rarity: common
    general-config:
      flags: [ ]
      targets:
        - bow
        - crossbow
      grindstoneable: true
      disabled-in-worlds: [ ]
      requirements:
        list: [ ]
        not-met-lore: [ ]
      conflicts:
        - buckshot
        - succession
        - tripleshot
    levels:
      - effects:
          - id: teleport
            args:
              cooldown: 2
            triggers:
              - projectile_hit
        conditions: [ ]
```

**Comment by WillFP** [comment]
Fixed in 8.30.0
